src/WebJobs.Script.WebHost/TokenExpiration/TokenExpirationService.cs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Properties;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.TokenExpiration
+{
+    internal class TokenExpirationService : BackgroundService, IDisposable
+    {
+        private readonly TaskCompletionSource<object> _standby = new();
+        private readonly IEnvironment _environment;
+        private readonly ILogger<TokenExpirationService> _logger;
+        private IDisposable _listener;
+        private CancellationTokenSource _cancellationTokenSource;
+        private Task _analysisTask;
+        private bool _analysisScheduled;
+
+        public TokenExpirationService(IEnvironment environment, ILogger<TokenExpirationService> logger, IOptionsMonitor<StandbyOptions> standbyOptionsMonitor)
+        {
+            _environment = environment;
+            _logger = logger;
+            if (standbyOptionsMonitor.CurrentValue.InStandbyMode)
+            {
+                _listener = standbyOptionsMonitor.OnChange(standbyOptions =>
+                {
+                    if (!standbyOptions.InStandbyMode && !_analysisScheduled)
+                    {
+                        _standby.TrySetResult(null);
+                    }
+                });
+            }
+            else
+            {
+                _standby.TrySetResult(null);
+            }
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+        {
+            await Task.Yield(); // force a yield, so our synchronous code does not slowdown startup
```

Comment:
If you want to retain your delay of 1 second you have right now (I may be missing context on why you have it), this can become:

```suggestion
            await Task.Delay(TimeSpan.FromSeconds(1), cancellationToken);
```

Both will achieve the same result of forcing a yield

Reply:
I don't think we actually need the delay of one second. That was old code from copying the `AssemblyAnalysisService`